stratum/hal/lib/common/phaldb_service.cc

```diff
@@ -0,0 +1,454 @@
+        break;
+      }
+      default:
+        attr_res = MAKE_ERROR(ERR_INVALID_PARAM) << "Invalid update query";
+    }
+
+    if (!attr_res.ok()) {
+        LOG(ERROR) << "Set update " << std::to_string(i) << " failed: "
+                   << attr_res.status().error_message();
+        // If we got an error set the top level status
+        status = attr_res.status();
+        results.push_back(attr_res.status());
+        continue;
+    }
+
+    auto path = attr_res.ConsumeValueOrDie();
+
+    // Create attribute path:val pair base on value type
+    switch (update.value().value_case()) {
+      case ::stratum::hal::phal::UpdateValue::kDoubleVal: {
+        attrs[path] = update.value().double_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kFloatVal: {
+        attrs[path] = update.value().float_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kInt32Val: {
+        attrs[path] = update.value().int32_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kInt64Val: {
+        attrs[path] = update.value().int64_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kUint32Val: {
+        attrs[path] = update.value().uint32_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kUint64Val: {
+        attrs[path] = update.value().uint64_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kBoolVal: {
+        attrs[path] = update.value().bool_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kStringVal: {
+        attrs[path] = update.value().string_val();
+        break;
+      }
+      case ::stratum::hal::phal::UpdateValue::kBytesVal: {
+        attrs[path] = update.value().bytes_val();
+        break;
+      }
+      default: {
+        attr_res = MAKE_ERROR(ERR_INVALID_PARAM) << "Unknown value type";
+        break;
+      }
+    }
+
+    // Push status onto results stack
+    results.push_back(attr_res.status());
+
+    // If we got an error set the top level status
+    if (!attr_res.ok()) {
+        status = attr_res.status();
+    }
+  }
+
+  // Do Set if we have no errors
+  if (status.ok()) {
+    // Note: all updates are passed down to PhalDB as one Set call
+    //       so we won't get individual status on each adapter attribute
+    //       update.
+    results = {};
+
+    // Do set for all attribute pairs
+    auto adapter =
+        absl::make_unique<::stratum::hal::phal::Adapter>(phal_interface_);
+    status = adapter->Set(attrs);
+  }
+
+  return ToGrpcStatus(status, results);
+}
+
+::grpc::Status PhalDBService::Subscribe(::grpc::ServerContext* context,
+    const ::stratum::hal::phal::SubscribeRequest* req,
+    ::grpc::ServerWriter<::stratum::hal::phal::SubscribeResponse>* stream) {
+
+  RETURN_IF_NOT_AUTHORIZED(auth_policy_checker_, PhalDBService,
+                           Subscribe, context);
+
+  ::util::Status status;
```

Comment:
Declare on first use, i.e. Line 358

Reply:
actually needs to be declared outside the if statement that follows.

Reply:
Are you sure? I see `result` being used, but not status.

Reply:
line 422, 
// Convert to grpc status and return
  return ToGrpcStatus(status, {});

status is set inside the "if statement" and then ToGrpcStatus converts it to a gRPC status and returns it outside the "if statement"?  Guess I could move the definition down to line 330 (i.e. first use).

Reply:
I missed that. Yes, right before the loop would be good.